source/s3_client.c

```diff
@@ -319,9 +322,22 @@ PyObject *aws_py_s3_client_new(PyObject *self, PyObject *args) {
         signing_config = &default_signing_config;
     }
 
+    struct aws_byte_cursor *network_interface_names = NULL;
+    int num_network_interface_names = 0;
+
+    if (network_interface_names_py != Py_None) {
+        if (!PyList_Check(network_interface_names_py)) {
+            PyErr_SetString(PyExc_TypeError, "Expected network_interface_names to be a list");
+            return NULL;
+        }
+        Py_ssize_t listSize = PyList_Size(network_interface_names_py);
+        num_network_interface_names = (size_t)listSize;
+    }
+
     struct s3_client_binding *s3_client = aws_mem_calloc(allocator, 1, sizeof(struct s3_client_binding));
 
     /* From hereon, we need to clean up if errors occur */
+    int result = AWS_OP_ERR;
 
     PyObject *capsule = PyCapsule_New(s3_client, s_capsule_name_s3_client, s_s3_client_capsule_destructor);
     if (!capsule) {
```

Comment:
after this if-check, there's still some manual cleanup and `return NULL` happening, change it to use the new `goto`

Reply:
Thanks, refactored the cleanup logic.